CMakeLists.txt

```diff
@@ -468,7 +468,7 @@ if(BUILD_LLVM_DWARF)
 endif()
 
 if(MIMALLOC_STATIC)
-  if(NOT(LINUX AND BUILD_STATIC_LIB) OR EMSCRIPTEN)
+  if(NOT(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND BUILD_STATIC_LIB) OR EMSCRIPTEN)
```

Comment:
When using the emscripten-releases build script I got a very strange behavior; namely that the `LINUX` variable was empty, but `CMAKE_SYSTEM_NAME` was set to `Linux` as expected. I couldn't figure out what was causing it. This change works around that.

Reply:
OK, I figure it out; the `LINUX` variable was introduced in CMake 3.25 but we support 3.16 and use 3.21 on our bots. This is the right way to do it on those versions, so I'll just update the commit message.